internal/resource/config/serversettings/systemkeys/server_settings_system_keys_data_source.go

```diff
@@ -0,0 +1,194 @@
+package serversettingssystemkeys
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/datasource/common/id"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &serverSettingsSystemKeysDataSource{}
+	_ datasource.DataSourceWithConfigure = &serverSettingsSystemKeysDataSource{}
+)
+
+// ServerSettingsSystemKeysDataSource is a helper function to simplify the provider implementation.
+func NewServerSettingsSystemKeysDataSource() datasource.DataSource {
+	return &serverSettingsSystemKeysDataSource{}
+}
+
+// serverSettingsSystemKeysDataSource is the resource implementation.
+type serverSettingsSystemKeysDataSource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type serverSettingsSystemKeysDataSourceModel struct {
+	Id       types.String `tfsdk:"id"`
+	Current  types.Object `tfsdk:"current"`
+	Previous types.Object `tfsdk:"previous"`
+	Pending  types.Object `tfsdk:"pending"`
+}
+
+// GetSchema defines the schema for the datasource.
+func (r *serverSettingsSystemKeysDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Describes the settings for server system keys.",
```

Comment:
This isn't really describing settings, it just returns the system keys themselves.